### DISCUSSION TITLE
Google fires "Unsecure browser" warning

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3851] [Q&A] [answered]
Hi! I`m trying to login in Youtube (Google)  manually in WinForms ChromiumWebBrowser component). However Im getting the following warning (screened below) which I cannot bypass. I`m using the same user agent as my main Chrome browser has: "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36". Any workarounds for forcing Google to recognize my browser component as secure browser?

![image](https://user-images.githubusercontent.com/73235464/137682920-e9ea83aa-2009-4fe0-9522-4e9220d055f1.png)

**Comment by amaitland** [comment, marked as answer]
> Starting in June 2019 Google will block logins from CEF based browsers to Google Services, this includes Gmail, Drive, Docs, see https://security.googleblog.com/2019/04/better-protection-against-man-in-middle.html?m=1 for further discussion see https://groups.google.com/a/chromium.org/d/msg/embedder-dev/STyM5ZNTHMM/POj1v_cqBgAJ

`Google` has blocked logins from all `embedded browsers`. This is not a flaw in `CefSharp/CEF`, it's that you can easily perform a man in the middle attack which is not something I can actually fix.

There are a few discussions on the topic already

- https://github.com/cefsharp/CefSharp/discussions/3506
- https://github.com/cefsharp/CefSharp/discussions/3538
- https://magpcss.org/ceforum/viewtopic.php?f=10&t=16764
- https://magpcss.org/ceforum/viewtopic.php?f=10&t=16717

> Any workarounds for forcing Google to recognize my browser component as secure browser?

`Google` haven't provided an official means to login to their service. Others embedded browsers are blocked as well including `Microsofts` new `Chrome` based embedded browser.

Others have tried switching the user agent with some success though no idea if that still actually works as this isn't something that I've tested myself.

**Reply by ukandrewc** [reply]
This works for me with GMail logon

`CefSettings.UserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:93.0) Gecko/20100101 Firefox/93.0"`
